json/src/ser.rs

```diff
@@ -489,44 +489,48 @@ impl<'a, W, F> ser::Serializer for MapKeySerializer<'a, W, F>
         Err(Error::Syntax(ErrorCode::KeyMustBeAString, 0, 0))
     }
 
-    fn serialize_isize(&mut self, _value: isize) -> Result<()> {
-        Err(Error::Syntax(ErrorCode::KeyMustBeAString, 0, 0))
+    fn serialize_isize(&mut self, value: isize) -> Result<()> {
+        self.serialize_i64(value as i64)
     }
 
-    fn serialize_i8(&mut self, _value: i8) -> Result<()> {
-        Err(Error::Syntax(ErrorCode::KeyMustBeAString, 0, 0))
+    fn serialize_i8(&mut self, value: i8) -> Result<()> {
+        self.serialize_i64(value as i64)
     }
 
-    fn serialize_i16(&mut self, _value: i16) -> Result<()> {
-        Err(Error::Syntax(ErrorCode::KeyMustBeAString, 0, 0))
+    fn serialize_i16(&mut self, value: i16) -> Result<()> {
+        self.serialize_i64(value as i64)
     }
 
-    fn serialize_i32(&mut self, _value: i32) -> Result<()> {
-        Err(Error::Syntax(ErrorCode::KeyMustBeAString, 0, 0))
+    fn serialize_i32(&mut self, value: i32) -> Result<()> {
+        self.serialize_i64(value as i64)
     }
 
-    fn serialize_i64(&mut self, _value: i64) -> Result<()> {
-        Err(Error::Syntax(ErrorCode::KeyMustBeAString, 0, 0))
+    fn serialize_i64(&mut self, value: i64) -> Result<()> {
```

Comment:
I think this should be:

```rust
try!(self.ser.writer.write_all(b"\""));
try!(self.ser.serialize_i64(value));
try!(self.ser.writer.write_all(b"\""));
Ok(())
```

The `serialize_str` method writes a double-quoted JSON string. We just need to write a byte directly into the output.